### DISCUSSION TITLE
Is there an endpoint documentation?

### DISCUSSION THREAD

**Discussion by ChinhasAngel** [NationalSecurityAgency/skills-service#1610] [Q&A] [answered]
Hello all.
Thanks for this project. My question is exist a documentation of the endpoints? I want a make review all endpoints.  
Greetings.

**Comment by sudo-may** [comment, marked as answer]
@ChinhasAngel Hope you are having a good day! The majority of the interactions with endpoints are encapsulated behind SkillTree's [Dashboard](https://skilltreeplatform.dev/dashboard/user-guide/) and SkillTree's [skills-client](https://skilltreeplatform.dev/skills-client/) libraries; both dashboard and skills-client libraries interact with the endpoints but they are not intended be called directly therefore they are not documented and can change in future releases. There are a handful of endpoints that are designed for direct consumption and their documentation can be found here: https://skilltreeplatform.dev/skills-client/endpoints.html

With that said the code for majority of endpoint resides in the [skills.controller](https://github.com/NationalSecurityAgency/skills-service/blob/master/service/src/main/java/skills/controller) package so it can be reviewed there and endpoints can be interacted with given the understand that their api could change in the future without notice. I hope this helps!
